integration/kotlinx-coroutines-guava/src/ListenableFuture.kt

```diff
@@ -38,81 +47,98 @@ public fun <T> CoroutineScope.future(
 ): ListenableFuture<T> {
     require(!start.isLazy) { "$start start is not supported" }
     val newContext = newCoroutineContext(context)
+    // TODO: It'd be nice not to leak this SettableFuture reference, which is easily blind-cast.
```

Comment:
A nice short way to do this, if you like: `return object: ListenableFuture<T> by future {}`.